pkg/batch/executor.go

```diff
@@ -11,19 +11,31 @@ import (
 // dispatching new ones would start blocking.
 const RequestBufferSize = 1 << 17
 
+type Executable interface {
+	Execute() (interface{}, error)
+}
+
+type Batchable interface {
+	Batched()
```

Comment:
Can you document this method?  I _think_ it's what the JavaScript world would call `OnBatched`.